kaldb/src/main/proto/metadata.proto

```diff
@@ -134,6 +134,9 @@ message DatasetMetadata {
 
   // List of partitions assigned to this service, and their effective times
   repeated DatasetPartitionMetadata partition_configs = 4;
+
+  // either same as name or a special value _all
+  string service_name = 5;
```

Comment:
I think `service_name_pattern` is a clearer option as mentioned above. This also won't be stored as straight regex, as it will need to be escaped specific to the flavor in use.